ddl/sequence_test.go

```diff
@@ -84,3 +84,70 @@ func (s *testSequenceSuite) TestCreateSequence(c *C) {
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Equals, "[planner:1142]CREATE command denied to user 'localhost'@'myuser' for table 'my_seq'")
 }
+
+func (s *testSequenceSuite) TestDropSequence(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use test")
+	s.tk.MustExec("drop sequence if exists seq")
+
+	// Test sequence is unknown.
+	s.tk.MustGetErrCode("drop sequence seq", mysql.ErrUnknownSequence)
+
+	// Test non-existed sequence can't drop successfully.
+	s.tk.MustExec("create sequence seq")
+	_, err := s.tk.Exec("drop sequence seq, seq2")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[schema:4139]Unknown SEQUENCE: 'test.seq2'")
+
+	// Test the specified object is not sequence.
+	s.tk.MustExec("create table seq3 (a int)")
+	_, err = s.tk.Exec("drop sequence seq3")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[ddl:1347]'test.seq3' is not SEQUENCE")
+
+	// Test schema is not exist.
+	_, err = s.tk.Exec("drop sequence unknown.seq")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[schema:4139]Unknown SEQUENCE: 'unknown.seq'")
+
+	// Test drop sequence successfully.
+	s.tk.MustExec("create sequence seq")
+	_, err = s.tk.Exec("drop sequence seq")
+	c.Assert(err, IsNil)
+	_, err = s.tk.Exec("drop sequence seq")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[schema:4139]Unknown SEQUENCE: 'test.seq'")
+
+	// Test drop table when the object is a sequence.
+	s.tk.MustExec("create sequence seq")
+	_, err = s.tk.Exec("drop table seq")
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[schema:1051]Unknown table 'test.seq'")
```

Comment:
Is it expected? MariaDB has different behavior.
```
mariadb root@172.17.0.3:test> create sequence s;
Query OK, 0 rows affected
Time: 0.019s
mariadb root@172.17.0.3:test> drop table s
Query OK, 0 rows affected
Time: 0.013s
```

Reply:
I've talked with wangcong about this diff. we think drop table shouldn't influence sequence object.